drivers/sensor/tdk/icm456xx/icm456xx.c

```diff
@@ -0,0 +1,1061 @@
+#define ICM456XX_CONFIG_COMMON(inst)                                                               \
+	IF_ENABLED(CONFIG_ICM456XX_TRIGGER,	\
+			(.gpio_int = GPIO_DT_SPEC_INST_GET_OR(inst, int_gpios, {0}),))     \
+			.accel_fs = DT_INST_ENUM_IDX(inst, accel_fs),	\
+			.accel_hz = DT_INST_ENUM_IDX(inst, accel_hz),	\
+			.accel_pwr_mode = DT_INST_ENUM_IDX(inst, power_mode),	\
+			.accel_avg = DT_INST_ENUM_IDX(inst, accel_avg),	\
+			.accel_filt_bw = DT_INST_ENUM_IDX(inst, accel_filt_bw_hz),	\
+			.apex = DT_INST_ENUM_IDX(inst, apex),                          \
+			.gyro_fs = DT_INST_ENUM_IDX(inst, gyro_fs),                     \
+			.gyro_hz = DT_INST_ENUM_IDX(inst, gyro_hz),                     \
+			.gyro_filt_bw = DT_INST_ENUM_IDX(inst, gyro_filt_bw_hz)
+
+/* Initializes the bus members for an instance on a SPI bus. */
+#define ICM456XX_CONFIG_SPI(inst)                                                                  \
+	{.bus.spi = SPI_DT_SPEC_INST_GET(inst, ICM456XX_SPI_CFG, 0),                               \
+	 .bus_io = &icm456xx_bus_io_spi,                                                           \
+	 .serif_type = UI_SPI3,                                                                    \
+	 ICM456XX_CONFIG_COMMON(inst)}
+
+/* Initializes the bus members for an instance on an I2C bus. */
+#define ICM456XX_CONFIG_I2C(inst)                                                                  \
+	{.bus.i2c = I2C_DT_SPEC_INST_GET(inst),                                                    \
+	 .bus_io = &icm456xx_bus_io_i2c,                                                           \
+	 .serif_type = UI_I2C,                                                                     \
+	 ICM456XX_CONFIG_COMMON(inst)}
+
+/*
+ * Main instantiation macro, which selects the correct bus-specific
+ * instantiation macros for the instance.
+ */
+#define ICM456XX_DEFINE(inst, name, whoami)                                                        \
+	static struct icm456xx_data icm456xx_data_##inst = {                                       \
+		.imu_name = name,                                                                  \
+		.imu_whoami = whoami,                                                              \
+	};                                                                                         \
+	static const struct icm456xx_config icm456xx_config_##inst = \
+			COND_CODE_1(DT_INST_ON_BUS(inst, spi),	\
+			(ICM456XX_CONFIG_SPI(inst)),	\
+			(ICM456XX_CONFIG_I2C(inst)));             \
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, icm456xx_init, NULL, &icm456xx_data_##inst,             \
+				     &icm456xx_config_##inst, POST_KERNEL,                         \
+				     CONFIG_SENSOR_INIT_PRIORITY, &icm456xx_driver_api);
+
+#define DT_DRV_COMPAT invensense_icm45686
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+DT_INST_FOREACH_STATUS_OKAY_VARGS(ICM456XX_DEFINE, /*INV_ICM45686_STRING_ID*/"ICM45686", 0xEE /*INV_IMU_WHOAMI*/);
+#endif
```

Comment:
These defines INV_ICM45686_STRING_ID and INV_ICM45686_WHOAMI are defined is TDK hal.

Reply:
Fixed